### PR TITLE
Implement $min and $max update operators

### LINKED ISSUE #223: Implement $min and $max update operators
Two more in-scope update operators are `$min` and `$max`:

* https://www.mongodb.com/docs/v6.0/reference/operator/update/min/
* https://www.mongodb.com/docs/v6.0/reference/operator/update/max/

One thing to note is that comparison needs to work beyond numeric values, such that:

* It is possible to compare, say, Strings (lexicographic sorting)
* It is also possible to compare values of different types (Numbers vs Strings) -- ordering by https://www.mongodb.com/docs/v6.0/reference/bson-type-comparison-order/#std-label-faq-dev-compare-order-for-BSON-types

We may limit ordering logic in case of JSON Objects to something simpler than what BSON rules dictate (as long as there is a stable ordering used).

For types that we do support, ordering (from lowest to highest) would be:

1. Missing / `null` (explicit)
2. Number
3. String
4. Object
5. Array
6. Boolean

to keep ordering the same as with Mongo/BSON.

This would basically leave out: Binary Data, Object Id, Date, Timestamp, Regexp (and min/max key markers) since these would expressed as one of JSON native types (mostly Strings).

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds `$min` and `$max` update operators

**Which issue(s) this PR fixes**:
Fixes #223

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by maheshrajamani — APPROVED** [review]

File: src/main/java/io/stargate/sgv2/jsonapi/util/JsonNodeComparator.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ * (NOTE: these are types we have -- MongoDB has more native types so this is a subset of BSON
+ * sorting definitions).
+ *
+ * <p>Within each type sorting is as usual for most types (Numbers, Strings, Booleans). Arrays use
+ * straight-forward element-by-element sorting (similar to Strings). The only more esoteric case are
+ * Objects, where sorting is by ordered fields, first comparing field name (String sort), if same,
+ * then recursively by value; and if first N fields the same, Object with more properties is sorted
+ * last.
+ */
+public class JsonNodeComparator implements Comparator<JsonNode> {
```
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/util/JsonNodeComparator.java:35]: Yes, was about to suggest just that, was thinking about it.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindAndUpdateIntegrationTest.java [hunk 2]
```diff
@@ -1805,4 +1804,146 @@ private void insertDoc(String docJson) {
...
+                          }
+                       }
+                       """);
+      String updateJson =
+          """
+                      {
+                        "updateOne": {
+                          "filter" : {"_id" : "update_doc_min"},
+                          "update" : {"$min" : {
+                              "min": 2,
```
Comment (vkarpov15) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindAndUpdateIntegrationTest.java:1829]: Would be nice to have an integration test that uses `$min` and `$max` with types that aren't numbers. For example, `$min: { foo: 'bar' }`. That should work given the `JsonNodeComparatorTest.java` tests, but I think it's worth checking.

Comment (tatu-at-datastax) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindAndUpdateIntegrationTest.java:1829]: Yeah. I did add unit tests for wider range given they are more compact, but yeah I could/should add at least one cross-type, and one non-number IT operation.

File: src/main/java/io/stargate/sgv2/jsonapi/util/JsonNodeComparator.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (devilshezi0) [reply src/main/java/io/stargate/sgv2/jsonapi/util/JsonNodeComparator.java:35]: > Cool. In memory sort made easy with this utility class.

[the remark this answers is not in the document]
Comment (devilshezi0) [reply src/main/java/io/stargate/sgv2/jsonapi/util/JsonNodeComparator.java:35]: > > Cool. In memory sort made easy with this utility class.
> 
>

**Pull request merged**